plugins/module_utils/utils.py

```diff
@@ -36,12 +37,26 @@ def do_differ(current, desired):
         return True
 
     for key, value in desired.items():
-        if value != current.get(key):
+        current_value = current.get(key)
+        # list elements are of the form {"type": "a", "name": "b"}
+        if (isinstance(value, list) and
+                all([isinstance(i, dict) and "name" in i and "type" in i for i in value])):
+            if type_name_dicts_differ(value, current_value):
+                return True
+        elif value != current_value:
             return True
 
     return False
 
 
+# sorts a list of dicts by "type" and "name" keys
+# and returns the result of comparison.
+def type_name_dicts_differ(a, b):
+    sorted_a = sorted(a, key=lambda x: (x["type"], x["name"]))
+    sorted_b = sorted(b, key=lambda x: (x["type"], x["name"]))
+    return sorted_a != sorted_b
+
+
```

Comment:
Should we move this to a separate role utils module (alongside with the custom comparator that I was fantasizing about)?